packages/fx-core/src/plugins/resource/frontend/dotnet/plugin.ts

```diff
@@ -202,6 +212,66 @@ export class DotnetPluginImpl implements PluginImpl {
   }
 
   public async localDebug(ctx: PluginContext): Promise<TeamsFxResult> {
+    if (isConfigUnifyEnabled()) {
+      ctx.envInfo.state.get(PluginInfo.pluginName)?.set(ConfigInfo.indexPath, PathInfo.indexPath);
+    } else {
+      ctx.localSettings?.frontend?.set(LocalSettingsFrontendKeys.TabIndexPath, PathInfo.indexPath);
+    }
+    return ok(undefined);
+  }
+
+  public async postLocalDebug(ctx: PluginContext): Promise<TeamsFxResult> {
+    const appSettingsPath = path.join(ctx.root, PathInfo.appSettingDevelopment);
+    let appSettings = await fs.readFile(appSettingsPath, "utf-8");
+
+    let clientId = AppSettingsPlaceholder.clientId;
+    let clientSecret = AppSettingsPlaceholder.clientSecret;
+    let tenantId = "";
+    let oauthAuthority = AppSettingsPlaceholder.oauthAuthority;
+    let botId = AppSettingsPlaceholder.botId;
+    let botPassword = AppSettingsPlaceholder.botPassword;
+
+    if (isConfigUnifyEnabled()) {
+      clientId =
+        ctx.envInfo.state.get(PluginNames.AAD)?.get(DependentPluginInfo.aadClientId) ??
+        AppSettingsPlaceholder.clientId;
+      clientSecret =
+        ctx.envInfo.state.get(PluginNames.AAD)?.get(DependentPluginInfo.aadClientSecret) ??
+        AppSettingsPlaceholder.clientSecret;
+      tenantId = ctx.envInfo.state.get(PluginNames.APPST)?.get(DependentPluginInfo.appTenantId);
+      oauthAuthority = tenantId
+        ? PathInfo.oauthHost(tenantId)
+        : AppSettingsPlaceholder.oauthAuthority;
+      botId =
+        ctx.envInfo.state.get(PluginNames.BOT)?.get(DependentPluginInfo.botId) ??
+        AppSettingsPlaceholder.botId;
+      botPassword =
+        ctx.envInfo.state.get(PluginNames.BOT)?.get(DependentPluginInfo.botPassword) ??
+        AppSettingsPlaceholder.botPassword;
+    } else {
+      clientId =
+        ctx.localSettings?.auth?.get(LocalSettingsAuthKeys.ClientId) ??
+        AppSettingsPlaceholder.clientId;
+      clientSecret =
+        ctx.localSettings?.auth?.get(LocalSettingsAuthKeys.ClientSecret) ??
+        AppSettingsPlaceholder.clientSecret;
```

Comment:
Nit: these lines are kind of verbose and within similar biz logic. Better to simplify them to improve readability if time allows.